src/main/kotlin/net/atos/zac/mail/MailMessageBuilder.kt

```diff
@@ -0,0 +1,50 @@
+package net.atos.zac.mail
+
+import jakarta.mail.Address
+import jakarta.mail.Message
+import jakarta.mail.Session
+import jakarta.mail.internet.MimeBodyPart
+import jakarta.mail.internet.MimeMessage
+import jakarta.mail.internet.MimeMultipart
+import jakarta.mail.internet.PreencodedMimeBodyPart
+import net.atos.zac.mail.model.Attachment
+import java.nio.charset.StandardCharsets
+import java.util.Date
+
+class MailMessageBuilder(
+    private var fromAddress: Address,
+    private var toAddress: Address,
+    private var replyToAddress: Address? = null,
+    private var mailSubject: String,
+    private var body: String,
+    private var attachments: List<Attachment>
+) {
+    fun build(mailSession: Session) = MimeMessage(mailSession).apply {
+        addHeader("Content-type", "text/HTML; charset=${StandardCharsets.UTF_8.name()}")
+        addHeader("format", "flowed")
+        addHeader("Content-Transfer-Encoding", "8bit")
+
+        setFrom(fromAddress)
+        setRecipients(Message.RecipientType.TO, arrayOf(toAddress))
+        replyTo = replyToAddress?.let { arrayOf(replyToAddress) }
+        subject = mailSubject
+        sentDate = Date()
+
+        setContent(
+            MimeMultipart().apply {
+                addBodyPart(
+                    MimeBodyPart().apply {
+                        setText(body, StandardCharsets.UTF_8.name(), "html")
+                    }
+                )
+                attachments.forEach {
+                    val messageBodyPart = PreencodedMimeBodyPart("base64").apply {
```

Comment:
Just wondering: is this standard in email attachments? Using base64 encoding I mean?

Reply:
Our `Attachment` gets the payload already base64 encoded. My idea here was to change as little as possible.

Since multi-part mime messages are base64 encoded, and Jakarta already supported pre-encoded parts with `Preencoded*` this felt like the easiest way to do.

Reply:
ok. Did you test if this works correctly (locally)?

Of course it would be great if we have actual integration tests that test some of the ZAC email functionality (with attachments even) but that is quite some work. We would most likely want to use one of those fake SMTP mocking  tools. For another day/PR for sure.

Reply:
Tested only that mail is in HTML format, but that was without attachments. 

Have to check why MailJet started return 535 autorization error

Reply:
![image](https://github.com/infonl/dimpact-zaakafhandelcomponent/assets/732222/4661ba10-671b-49fb-8321-ca4e2ed50660)

Works!